wasm/src/debug_service.rs

```diff
@@ -35,15 +36,24 @@ impl DebugService {
 
     pub fn load_source(
         &mut self,
-        path: String,
-        source: String,
+        // can't wasm_bindgen [string; 2] or (string, string)
+        // so we have to manually assert length of the interior
+        // array and the content type in the function body
+        // `sources` should be Vec<[String; 2]> though
+        sources: Vec<js_sys::Array>,
         target_profile: String,
         entry: Option<String>,
     ) -> String {
-        let source_map = SourceMap::new(
-            [(path.into(), source.into())],
-            entry.as_deref().map(|value| value.into()),
-        );
+        let sources = sources.into_iter().map(|js_arr| {
+            // map the inner arr elements into (String, String)
+            let elem_0 = js_arr.get(0).as_string();
+            let elem_1 = js_arr.get(1).as_string();
+            (
+                Arc::from(elem_0.unwrap_or_default()),
+                Arc::from(elem_1.unwrap_or_default()),
+            )
+        });
+        let source_map = SourceMap::new(sources, entry.as_deref().map(|value| value.into()));
```

Comment:
If you have a moment it'd be helpful to extract this conversion into a helper function, since it'll be used by the Compiler methods (`run()` etc) as well.